cosmwasm/enclaves/shared/block-verifier/src/validator_whitelist.rs

```diff
@@ -0,0 +1,54 @@
+use tendermint_light_client_verifier::types::UntrustedBlockState;
+
+#[cfg(not(feature = "production"))]
+const WHITELIST_FROM_FILE: &str = include_str!("../fixtures/validator_whitelist.txt");
+#[cfg(feature = "production")]
+const WHITELIST_FROM_FILE: &str = include_str!("../fixtures/validator_whitelist_prod.txt");
+
+#[cfg(not(feature = "production"))]
+const VALIDATOR_THRESHOLD: usize = 1;
+
+#[cfg(feature = "production")]
+const VALIDATOR_THRESHOLD: usize = 5;
+
+lazy_static::lazy_static! {
+    static ref VALIDATOR_WHITELIST: ValidatorList = ValidatorList::from_str(WHITELIST_FROM_FILE);
+}
+
+#[derive(Debug, Clone)]
+struct ValidatorList(pub Vec<String>);
+
+impl ValidatorList {
+    fn from_str(list: &str) -> Self {
+        let addresses: Vec<String> = list.split(',').map(|s| s.to_string()).collect();
+        Self(addresses)
+    }
+
+    // use for tests
+    #[allow(dead_code)]
+    fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    fn contains(&self, input: &String) -> bool {
+        self.0.contains(input)
+    }
+}
+
+pub fn whitelisted_validators_in_block(untrusted_block: &UntrustedBlockState) -> bool {
+    untrusted_block.validators.validators().iter().filter(|&a| {VALIDATOR_WHITELIST.contains(&a.address.to_string())}).count() >= VALIDATOR_THRESHOLD
```

Comment:
Could someone submit a block where a validator appears many times, which then causes one whitelisted validator to be able to pass the light client verification? maybe we need to do a unique()

Reply:
Interesting idea. Although This would change the block hash and therefore the signatures won't verify correctly.

Reply:
@eshelB check this out: https://github.com/scrtlabs/tendermint-rs/blob/4019839ff894e531fbe4187f8461e4395d203220/light-client-verifier/src/operations/voting_power.rs#L128-L132